File: engine/src/main/java/org/jesse/game/world/entity/npc/impl/slayer/superior/impl/DreadbornAraxyte.java
package org.jesse.game.world.entity.npc.impl.slayer.superior.impl;


import org.jesse.game.world.entity.Entity;
import org.jesse.game.world.entity.Location;
import org.jesse.game.world.entity.masks.Animation;
import org.jesse.game.world.entity.masks.Hit;
import org.jesse.game.world.entity.masks.HitType;
import org.jesse.game.world.entity.npc.NPC;
import org.jesse.game.world.entity.npc.combat.CombatScript;
import org.jesse.game.world.entity.npc.combatdefs.AttackType;
import org.jesse.game.world.entity.npc.impl.slayer.superior.SuperiorNPC;
import org.jesse.game.world.entity.player.Player;
import org.jesse.game.world.entity.player.action.combat.CombatUtilities;

import static org.jesse.game.npc.ids.NpcId.DREADBORN_ARAXYTE;

public class DreadbornAraxyte extends SuperiorNPC implements CombatScript {

    public DreadbornAraxyte(Player owner, NPC root, Location spawnLocation) {
        super(owner, root, DREADBORN_ARAXYTE, spawnLocation);
    }

    @Override
    public int attack(Entity target) {
        if (target == null) {
            return 1;  // nothing to smack
        }

        // play the attack animation
        this.setAnimation(new Animation(9140));

        // roll your random max hit
        int damage = CombatUtilities.getRandomMaxHit(
                this,
                31,
                AttackType.MELEE,
                target
        );

        // schedule the hit on them
        Hit hit = new Hit(this, damage, HitType.DEFAULT);
        target.scheduleHit(this, hit, 0);

        return 4;  // attack delay ticks
    }
}

File: engine/src/main/kotlin/org/jesse/game/content/araxxor/araxytes/impl/DreadbornAraxyte.kt
package org.jesse.game.content.araxxor.araxytes.impl

import org.jesse.game.content.damage
import org.jesse.game.content.hit
import org.jesse.game.content.seq
import org.jesse.game.world.entity.Entity
import org.jesse.game.world.entity.Location
import org.jesse.game.world.entity.npc.NPC
import org.jesse.game.npc.ids.*
import org.jesse.game.world.entity.npc.combat.CombatScript
import org.jesse.game.world.entity.npc.combatdefs.AttackType
import org.jesse.game.world.entity.npc.impl.slayer.superior.SuperiorNPC
import org.jesse.game.world.entity.player.Player
import org.jesse.game.world.entity.player.action.combat.CombatUtilities

/**
 * @author Glabay | Glabay-Studios
 * @project near-reality-server
 * @social Discord: Glabay
 * @since 2024-10-20
 */
class DreadbornAraxyte(
    spawnLocation: Location,
    owner: Player,
    root: NPC
): SuperiorNPC(owner, root, DREADBORN_ARAXYTE, spawnLocation), CombatScript {

    override fun attack(target: Entity?): Int {
        target ?: return 1
        this seq 9140
        val damage = CombatUtilities.getRandomMaxHit(this, 31, AttackType.MELEE, target)
        target.scheduleHit(this, this hit target damage damage, 0)
        return 6
    }
}